service/history/api/startworkflow/api.go

```diff
@@ -318,6 +316,8 @@ func (s *Starter) handleConflict(
 	request := s.request.StartRequest
 	currentWorkflowRequestIDs := currentWorkflowConditionFailed.RequestIDs
 	if requestIDInfo, ok := currentWorkflowRequestIDs[request.GetRequestId()]; ok {
+		metrics.StartWorkflowRequestDeduped.With(s.metricsHandler).Record(1)
```

Comment:
I wonder if it's better to emit the metric based on the StartOutcome at a higher level.

Reply:
Hm, if I move to a higher level, I'd have to init metricsHandler at all times, which would be basically not doing lazy init for metricsHandler.
Also, the other cases are successful started, reused and error. With this new metric tracking dedup, we are only missing the distinction between actual start and reused.